src/CSET/operators/collapse.py

```diff
@@ -163,10 +181,34 @@ def collapse_by_hour_of_day(
     To apply this operator successfully there must only be one time dimension.
     Should a MultiDim exception be raised the user first needs to apply the
     collapse operator to reduce the time dimensions before applying this
-    operator.
+    operator. If multi_case is true the collapse_by_lead_time operator is
+    applied and performs this step.
     """
     if method == "PERCENTILE" and additional_percent is None:
         raise ValueError("Must specify additional_percent")
+    if (
+        isinstance(cube, iris.cube.Cube)
+        and is_time_aggregatable(cube)
+        and not multi_case
+    ):
+        raise TypeError(
+            "multi_case must be true for a cube containing two time dimensions"
+        )
+    if (
+        isinstance(cube, iris.cube.Cube)
+        and not is_time_aggregatable(cube)
+        and multi_case
+    ):
+        raise TypeError(
+            "multi_case must be false for a cube containing one time dimension"
+        )
+    if isinstance(cube, iris.cube.CubeList) and not multi_case:
+        raise TypeError("multi_case must be true for a CubeList")
```

Comment:
```suggestion
    multi_case = isinstance(cube, iris.cube.CubeList) or is_time_aggregatable(cube)
```
We can automate the check for `multi_case`.

I think this captures all of the conditions required. If it does then remove the `multi_case` argument from the function signature and update the docs.

Reply:
Automation of multi_case identification loses traceability, could lead to wrong implementation (i.e. different variables added in a cube list and not filtered properly), and "black box" mentality. For these reasons the user should specify if it is multi_case to ensure full documentation in the evaluation process.

Reply:
Also given that the default is set to True these are checks to indicate whether it needs to be changed, in most instances it should not need to be changed by the user.